Updating history view to add changes info (#6608)
mendix/docs: content/en/docs/refguide/modeling/menus/version-control-menu/history-dialog.md

title: "History"
url: /refguide/history-dialog/
weight: 50
tags: ["studio pro", "commit", "history", "select revision", "commit history"]
---
## 1 Introduction

Use the **History** dialog box to look at all the changes that have been committed to a development line of an app. You can open this dialog box from the **Version Control** > **History…** menu item. This dialog box is also used when selecting a revision to merge from, create a branch line off, or to create a deployment archive from.

## 2 Revisions

{{< figure src="/attachments/refguide/modeling/menus/version-control-menu/history-dialog/revisions.png" alt="Revisions Image" >}}

Each line in the grid represents a revision in the repository. For each revision the following information is shown, more information about each of these values is available below:

* [Revision](#revision)
* [Changes](#changes)
* [Author](#author)
* [Date](#date)
* [Time](#time)
* [Message](#message)

By selecting a revision in this grid, the tabs below the grid will be filled with information about this revision.

### 2.1 Revision{#revision}

> A unique alphanumeric identifier of the revision in the repository. The revision identifier is unique for a whole repository.

Reviewer: Should we mention here that this is the Git commit id (hash)?

Author: Honestly not sure, would user need this information?

Reviewer: It might help if they are trying to do something outside Studio Pro?
@mikita-haurykau - what do you think?

Reviewer 2: Yes, that's good to add @MarkvanMents.

Reviewer: Updated in dae3dda0e6eaf1e347ad4c1ee7d5cabdb40eba9c

### 2.2 Changes{#changes}

A visual summary of the changes in the selected revision, each icon represents different type of change:

| Icon | Change Type | Notes |
| --- | --- | --- |
| {{< figure src="/attachments/refguide/modeling/menus/version-control-menu/history-dialog/MergeRevision.png" width=32 alt="Merge commit icon" >}} | Merge commit | Indicates that the commit is a merge of two different commits.<br/>Hovering over will show both the parent [revisions](#revision). |

> | {{< figure src="/attachments/refguide/modeling/menus/version-control-menu/history-dialog/RevChangesModel.png" width=32 alt="Model changes icon" >}} | Model changes | Indicates that there were some changes made to the domain model.<br/>For example, a new entity was added, an attribute renamed,  or an association changed. |

Reviewer: Are you sure this is only the Domain Model. Or is it changes to the whole app model?

Author: Not 100%, @mikita-haurykau would probably know.

Reviewer 2: This is not limited to the Domain Model, but 'model changes' in general. 

In the "for example" section I'd suggest mentioning Domain Model, Microflows and Pages - as that gives a good idea what is part of the model.

Reviewer: Updated in dae3dda0e6eaf1e347ad4c1ee7d5cabdb40eba9c

| {{< figure src="/attachments/refguide/modeling/menus/version-control-menu/history-dialog/RevChangesDisk.png" width=32 alt="Changes on disk icon" >}} | Changes on disk | Indicates that there were changes on disk.<br/>For example, a file was added or removed. |
| {{< figure src="/attachments/refguide/modeling/menus/version-control-menu/history-dialog/RevChangesVersion.png" width=32 alt="Studio Pro changes icon" >}} | Studio Pro changes | Indicates that there was a change to the Studio Pro version used to work with the app.<br/>For example, when upgrading the app from a previous version to the latest one. |

> | {{< figure src="/attachments/refguide/modeling/menus/version-control-menu/history-dialog/Solution.png" width=32 alt="Solution version changes icon" >}} | Solution version changes| Indicates that there was an upgrade of based on solution. |

Reviewer: Can we be a bit clearer what this means. Do you mean that the app now has a new version of a solution module? Perhaps we should point at the solutions module documentation https://docs.mendix.com/refguide/configure-add-on-and-solution-modules/.

Is this icon also shown if you add a new solution module, or delete an existing one?

Author: It means that the add on solution was updated to new version,  so publisher created new solution version and user upgraded it in his application. Pointing to doc seems great idea to better explain users what it is about exactly. 

From what I see only upgrading the version has this icon, adding solution doesn't.

Reviewer: How about:
```suggestion
| {{< figure src="/attachments/refguide/modeling/menus/version-control-menu/history-dialog/Solution.png" width=32 alt="Solution version changes icon" >}} | Solution version changes | Indicates that a [solution module](https://docs.mendix.com/refguide/configure-add-on-and-solution-modules/) used in the app was updated. |
```

Hover over this column in Studio Pro to get a textual summary of the changes.

### 2.3 Author{#author}

The person who committed the change.

### 2.4 Date{#date}

The date the revision was created, it can have one of the following values:

* **Today** - shown when revision was created today.
* **Yesterday** - shown when revision was created yesterday.
* the date - shown when the revision was created earlier than yesterday.

### 2.5 Time{#time}

The time the revision was created.

### 2.6 Message{#message}

> The user-entered message saved with the commit. If the message is long, you can read it more easily in the Message tab page, below.

Reviewer: I wonder if we should be more neutral about the commit message. Might the message be generated automatically by a merge or when resolving conflicts?

Or perhaps by an API in the future?

Author: As we learned lately we have flows in which it is not only user created.

Reviewer: So can we just remove the "user-entered" like this?
```suggestion
The message saved with the commit. If the message is long, you can read it more easily in the Message tab page, below.
```

Reviewer 2: Yes, let's do that @MarkvanMents.
